elements/lisk-utils/src/math/q.ts

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable no-bitwise */
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { MAX_FRAC, ONE, Q_OPERATION, TWO } from './constants';
+
+export class Q {
+	private readonly _base: bigint;
+	private _val: bigint;
+
+	public constructor(val: number | bigint | Buffer, base: number | bigint = 96) {
+		this._base = BigInt(base);
+		if (this._base > MAX_FRAC) {
+			throw new Error('Base exceeds max fraction allowed.');
+		}
+		if (typeof val === 'number') {
+			this._val = this._numberToQ(val);
+			return;
+		}
+		if (typeof val === 'bigint') {
+			this._val = val * TWO ** this._base;
+			return;
+		}
+		this._val = BigInt(`0x${val.toString('hex')}`);
+	}
+
+	public add(n: Q) {
+		if (this._base !== n._base) {
+			throw new Error('Base of input does not match.');
+		}
+		this._val += n._val;
+		return this;
```

Comment:
Q96 is needed on SDK for DPoS, so we added as part of library.

good point about mutability, since it's number, probably i shouldve kept it as @Incede did. Ill update to be immutable